pylint/checkers/typecheck.py

```diff
@@ -451,10 +451,10 @@ def _emit_no_member(
             except astroid.MroError:
                 return False
             if metaclass:
-                if _enum_has_attribute(owner, node):
-                    return False
                 # Renamed in Python 3.10 to `EnumType`
-                return metaclass.qname() in {"enum.EnumMeta", "enum.EnumType"}
+                if metaclass.qname() in {"enum.EnumMeta", "enum.EnumType"}:
+                    return not _enum_has_attribute(owner, node)
```

Comment:
Outside the scope of this PR, since status quo, but this looks fragile against importing `enum` under an alias, which @DanielNoord was just looking into at https://github.com/PyCQA/astroid/pull/1587. Probably worth a follow-up issue.

Reply:
Yeah, we should probably revisit this when we bump to `2.12`.